lib/download.js

```diff
@@ -71,49 +72,75 @@ module.exports = function (url, version, dest, cb) {
+            .on('end', function() {
+              console.log(url + ' is now extracted to ' + dest);
+              var err;
+              var home = homedir();
+              var icDir = path.join(home, 'oracle-instant-client');
+              if (!fs.existsSync(icDir)) {
+                try {
+                  fse.copySync(path.join(dest, 'instantclient'), icDir,
+                      {preserveTimestamps: true});
+                  console.log('Oracle instant client is installed to %s', icDir);
+                } catch (e) {
+                  err = e;
+                  console.error(e);
+                }
+              } else {
+                console.warn('Oracle instant client found at %s', icDir);
+              }
+              cb && cb(err);
+            }).on('error', function(error) {
+          cb && cb(error);
+        });
       }
     });
-
-    src.pipe(gunzip).pipe(tar.Extract({path: dest}))
-      .on('end', function() {
-        console.log(url + ' is now extracted to ' + dest);
-        var err;
-        var home = homedir();
-        var icDir = path.join(home, 'oracle-instant-client');
-        if (!fs.existsSync(icDir)) {
-          try {
-            fse.copySync(path.join(dest, 'instantclient'), icDir,
-              {preserveTimestamps: true});
-            console.log('Oracle instant client is installed to %s', icDir);
-          } catch (e) {
-            err = e;
-            console.error(e);
-          }
-        } else {
-          console.warn('Oracle instant client found at %s', icDir);
-        }
-        cb && cb(err);
-      }).on('error', function(error) {
-      cb && cb(error);
-    });
   }
 };
 
+function checkPrereqsAndInstallOracledb(url, cb) {
+  console.log('Pre-packaged file ' +  url  + ' doesn\'t exist.' + ' Checking if user has installed pre-requisites offline as per instructions in https://github.com/oracle/node-oracledb/blob/master/INSTALL.md' + '\n');
+  //can not find the oracle instance client prepackaged tarball.  Check if user has installed offline oracle instance client
+  //libraries and necessary pre-reqs as per instructions https://github.com/oracle/node-oracledb/blob/master/INSTALL.md
+  if (process.env.OCI_LIB_DIR != null && process.env.OCI_INC_DIR != null) { // check the oracle pre-req variables are set
+    //save current working dir
+    var currentDir = process.cwd();
+    //when loopback-connector-oracle is installed from apiconnect project, loopback-oracle-installer is installed
+    //at the same level as loopback-connector-oracle. oracledb should also need to be installed at the same level.
+    //traverse to the apiconnct project level and install oracledb. This will also work in loopback-connector-oracle dev env
+    //where loopback-oracle-installer is installed as dependent module to loopback-connector-oracle. In this case, oracledb
+    //is installed as dependent module of  loopback-connector-oracle.
+    var dirName =  path.join(__dirname, '..', '..', '..');
```

Comment:
We probably should use `require.resolve('loopback-connector-oracle/package.json')` to locate the oracle connector module.